memory/src/main/scala/filodb.memory/format/vectors/DeltaDeltaVector.scala

```diff
@@ -253,6 +257,20 @@ object DeltaDeltaConstDataReader extends LongVectorDataReader {
       out
     }
   }
+
+  def changes(vector: BinaryVectorPtr, start: Int, end: Int): Double = {
+    require(start >= 0 && end < length(vector), s"($start, $end) is out of bounds, length=${length(vector)}")
+    var prev = Double.NaN
+    var changes = 0
```

Comment:
So, this method belongs to a `ConstDeltaDeltaVector`.  This is a vector where everything is on a line slope already, so there's no point to iterate over each value.  Instead, there are two possible cases here:
a) slope == 0.  In this case, changes will always return 0, as the value is always constant - it never changes. (Or 1, I guess)
b) slope != 0.  In this case, every value is guaranteed to change - because the values are calculated.  Thus, changes should return the number of values in the range. 

The above would be much much faster and be optimal, no need to iterate on every value.